### DISCUSSION TITLE
FUTURE PROOF

### DISCUSSION THREAD

**Discussion by JonnyHurricane** [apohl79/audiogridder#1240] [Ideas]
I am 60. I started making music on computers in 1984. First of all, this entire architecture is genius. My studio started off as a bunch of synths and a Commodore 64. Over the years I've increasingly worked in the box. My studio is now an M1 Mac Mini and my old computer an i7 Mac Mini, a big screen that can accept two inputs and two MOTU interfaces that are bridged.

It recently occurred to me when I ran out of CPU on the M1 that I could just run Ableton on two machines and use Link to spread the load. Then Logic added Link support and suddenly I could run Ableton and Logic together in sync giving me an option to choose the best features from each of those worlds.

This was useful but didn't exactly work as hoped. Then I discovered BitWig. So I'm running BitWig like a tape machine and Ableton, FL Studio or MainStage as my "samplers" and "synths".

Anyway. The great thing about working in the box. Total recall. The worst thing about working in the box. UPGRADES!!!

The first programs that really allowed me to "save everything" in one place were early versions of Cubase VST with its dubious reverb and synth. ReBirth. FLStudio. then Logic, Pro Tools, Ableton etc. Seemed like magic at the time I could write a song, record it mix it, come back to it later. 

Or can I?

NO .. I CAN'T :-(

About ten years ago I decided Ableton provided the most stable platform for me. It was cross platform so I could load my PC sessions on a Mac. Or thought I could till I got another Mac.

The PC has by far the largest number of weird and wonderful VST plugins. So of course I tried them. Many of them. My sessions make absolutely no sense at all on a Mac.

Then KVR had some interesting challenges. So I have a load of weird and wonderful plugins on my Mac.

But VST plugins don't work in Logic. GAH!!!

And AU plugins don't work in BitWig. GAH!!!

And I sometimes edit things on my wife's PC. 

Then .. Ableton went 64Bit only ..

Then .. everything went M1

So more time trying to work out what works what doesn't work paying for upgrades etc when all I want to do is make music.

Here is why your program is the most incredible solution to my problem.

I can build a machine with all my current plugins on it and access it using Audiogridder and treat it like a piece of legacy hardware. As a server it won't need to be upgraded as long as I have an AU/VST client in my DAW that can access it. 

Brilliant. Or brilliant for as long as Audiogridder is around. 

So here is my suggestion and request.

1. Build a 32 bit server for Mac and PC so I can access 32bit plugins (there are some really amazing free ones out there which are no longer supported which I would love to integrate with my current DAW)

2. Make sure that subsequent versions of the client plugin are compatible with earlier versions of the server. 

By this I mean when you release version 5 of AudioGridder make sure there's a legacy option to search for older servers on the network and to communicate with them. So if I build a version today the plug-in you release in twenty years time can still talk to it.

Finally, I think you should talk to the people at BitWig about CLAP and create a CLAP version of the plug-in (not the server).

In the same way Ableton Link has now found it's way into major DAWs, you could create a relationship with a DAW manufacturer so end users have AudioGridder built into their DAW instead of as a plug-in?

**Comment by apohl79** [comment]
Thanks for your suggestions!

For 32 bit support there's already a feature request. 

Regarding backwards compat between plugins and server versions. It is a good point. It was not possible between v1.1 and v1.2 though, but I agree with that it should be a priority for future versions. What you can usually do is downgrading the plugins, if you have to use an old server.
